nemo/collections/nlp/models/language_modeling/megatron_glue_model.py

```diff
@@ -0,0 +1,279 @@
+        raise NotImplementedError("Testing is not supported for GLUE because the test data does not have labels. To evaluate on the validation dataset, call trainer.validate(model)")
+
+    def test_epoch_end(self, outputs):
+        raise NotImplementedError("Testing is not supported for GLUE because the test data does not have labels. To evaluate on the validation dataset, call trainer.validate(model)")
+
+    def build_train_valid_test_datasets(self, validation_only=False):
+        logging.info('Building GLUE datasets.')
+        self._validation_ds = TextToTextGLUEDataset(
+            self.cfg.data.validation_ds.file_path,
+            task_name=self.cfg.data.validation_ds.task_name,
+            tokenizer=self.model.tokenizer,
+            max_seq_length=self.cfg.data.validation_ds.max_seq_length,
+        )
+        if validation_only:
+            return None, self._validation_ds
+        self._train_ds = TextToTextGLUEDataset(
+            self.cfg.data.train_ds.file_path,
+            task_name=self.cfg.data.train_ds.task_name,
+            tokenizer=self.model.tokenizer,
+            max_seq_length=self.cfg.data.train_ds.max_seq_length,
+        )
+        logging.info(f'Length of train dataset: {len(self._train_ds)}')
+        logging.info(f'Length of val dataset: {len(self._validation_ds)}')
+        logging.info(f'Finished building T5 datasets.')
+        return self._train_ds, self._validation_ds
+
+    def build_pretraining_data_loader(self, dataset, batch_size, shuffle, num_workers, pin_memory):
+        """Buld dataloader given an input dataset."""
+
+        if dataset is None:
+            return None
+
+        rank = parallel_state.get_data_parallel_rank()
+        world_size = parallel_state.get_data_parallel_world_size()
+        sampler = torch.utils.data.distributed.DistributedSampler(
+            dataset, num_replicas=world_size, rank=rank, shuffle=shuffle
+        )
+
+        # Data loader. Note that batch size is the per GPU batch size.
+        return torch.utils.data.DataLoader(
+            dataset,
+            collate_fn=dataset.collate_fn,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            drop_last=False,
+        )
+
+    def setup(self, stage=None):
+        if stage == 'predict':
+            return
+
+        # NOTE: PTL uses the same stage string "test" for both testing and validation.
+        self.build_train_valid_test_datasets(validation_only=stage == 'validate')
+        self.setup_validation_data()
+        if stage == 'validate':
+            return
+        self.setup_training_data()
+
+    def setup_training_data(self):
+        self._train_dl = self.build_pretraining_data_loader(
+            self._train_ds,
+            self.cfg.data.train_ds.batch_size,
+            shuffle=True,
+            num_workers=self.cfg.data.train_ds.num_workers,
+            pin_memory=True,
```

Comment:
for setup_*_data methods shuffle and pin_memory are hardcoded not set via config?